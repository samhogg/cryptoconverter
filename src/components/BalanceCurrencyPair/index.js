import React from "react";
import PropTypes from "prop-types";
import currency from "../../config/currency";
import {
  BalanceItem,
  BalanceTicker,
  BalanceQuantity,
  BalancePrefix
} from "./styledComponents";

const BalanceCurrencyPair = ({ ticker, quantity }) => (
  <BalanceItem>
    <BalanceTicker>{ticker}</BalanceTicker>
    <BalanceQuantity>
      <BalancePrefix>{currency[ticker].prefix}</BalancePrefix>
      {quantity.toLocaleString("en-US", {
        minimumFractionDigits: currency[ticker].dp
      })}
    </BalanceQuantity>
  </BalanceItem>
);

BalanceCurrencyPair.propTypes = {
  ticker: PropTypes.string.isRequired,
  quantity: PropTypes.number.isRequired
};

export default BalanceCurrencyPair;
