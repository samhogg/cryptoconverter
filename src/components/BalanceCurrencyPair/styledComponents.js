import styled from "styled-components";

const BalanceItem = styled.li`
  padding: 0;
  margin: 0;
  flex: 1;
`;

const BalanceTicker = styled.div`
  text-transform: uppercase;
  font-size: 1rem;
  color: #666;
`;

const BalanceQuantity = styled.div`
  font-weight: 500;
  font-size: 1.5rem;
`;

const BalancePrefix = styled.span`
  padding-right: 0.2rem;
  color: #666;
  font-size: 1.3rem;
  font-weight: 400;
`;

export { BalanceItem, BalanceTicker, BalanceQuantity, BalancePrefix };
