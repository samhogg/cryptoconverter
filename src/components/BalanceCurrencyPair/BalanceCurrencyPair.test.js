import React from "react";
import BalanceCurrencyPair from ".";
import renderer from "react-test-renderer";

it("Renders with zero balances", () => {
  const tree = renderer
    .create(<BalanceCurrencyPair quantity={0} ticker="usd" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Renders with non zero balances", () => {
  const tree = renderer
    .create(<BalanceCurrencyPair ticker="usd" quantity={40000} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
