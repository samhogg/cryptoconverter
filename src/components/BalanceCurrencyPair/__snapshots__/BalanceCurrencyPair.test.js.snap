// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with non zero balances 1`] = `
<li
  className="styledComponents__BalanceItem-fDtCqf gzMZya"
>
  <div
    className="styledComponents__BalanceTicker-dDjLel bpTZSY"
  >
    usd
  </div>
  <div
    className="styledComponents__BalanceQuantity-kFtsKU jMHtFP"
  >
    <span
      className="styledComponents__BalancePrefix-fCshfk eIVmaR"
    >
      $
    </span>
    40,000.00
  </div>
</li>
`;

exports[`Renders with zero balances 1`] = `
<li
  className="styledComponents__BalanceItem-fDtCqf gzMZya"
>
  <div
    className="styledComponents__BalanceTicker-dDjLel bpTZSY"
  >
    usd
  </div>
  <div
    className="styledComponents__BalanceQuantity-kFtsKU jMHtFP"
  >
    <span
      className="styledComponents__BalancePrefix-fCshfk eIVmaR"
    >
      $
    </span>
    0.00
  </div>
</li>
`;
